lib/unified_health_data/adapters/oracle_health_prescription_adapter.rb

```diff
@@ -179,13 +272,93 @@ def extract_refill_remaining(resource)
       end
 
       # Extracts and normalizes MedicationRequest status to VistA-compatible values
+      # Checks for successful submitted refills based on Task resources
       #
       # @param resource [Hash] FHIR MedicationRequest resource
+      # @param task_data [Array<Hash>] Array of task resource hashes
+      # @param dispenses_data [Array<Hash>] Array of dispense data
       # @return [String] VistA-compatible status value
-      def extract_refill_status(resource)
+      def extract_refill_status(resource, task_data = [], dispenses_data = [])
+        # Check if there's a successful submitted refill
+        if has_recent_submitted_refill?(task_data, dispenses_data)
+          return 'submitted'
+        end
+
         normalize_to_legacy_vista_status(resource)
       end
 
+      # Checks if prescription has a recent successful refill submission
+      # Conditions:
+      # 1. Most recent task status = 'requested' (not 'failed')
+      # 2. Task executionPeriod.start within last 30 days
+      # 3. No subsequent MedicationDispense created (check meta.lastUpdated and versionId)
+      #
+      # @param task_data [Array<Hash>] Array of task resource hashes
+      # @param dispenses_data [Array<Hash>] Array of dispense data
+      # @return [Boolean] True if has recent submitted refill
+      def has_recent_submitted_refill?(task_data, dispenses_data)
```

Comment:
This looks a lot like duplicate functionality from ln 185? We just need to parse a few attributes from the contained Task resources to make a decision about whether to set refill_status to submitted and to populate the existing refill submitted date attribute.